### PR TITLE
Update with prelim release of nrf52-hal

### PR DESCRIPTION
**Pull request by jamesmunns**
Once https://github.com/nrf-rs/nrf52-hal/pull/32 lands, I will move this to just `0.6.0`. Or I might impl the SPI traits. Probably the former :)

### CODE REVIEW

**Comment by jamesmunns** [comment]
Note: Stable fails because I introduced `edition = "2018"`. We can park this for ~6 weeks if you like, or I can mess with the test matrix. Let me know what you think :)

**Comment by hannobraun** [comment]
Thank you James, this looks good to me! Did you verify that the examples in dwm1001 still work?

> Once [nrf-rs/nrf52-hal#32](https://github.com/nrf-rs/nrf52-hal/pull/32) lands, I will move this to just `0.6.0`. Or I might impl the SPI traits. Probably the former :)

Depending on `0.6.0` is fine. Doing everything required to depend only on embedded-hal is on my task list for the next few weeks, so it's going to happen anway, sooner or later.

> Note: Stable fails because I introduced `edition = "2018"`. We can park this for ~6 weeks if you like, or I can mess with the test matrix. Let me know what you think :)

Let's park this for a few weeks. This crate has been working on stable for a while, and I don't want to needlessly break consumers of `master`. Since the first crates.io release is not out yet, `master` is pretty much the only means to depend on this crate.

If we end up doing more work that conflicts with this PR before the 2018 edition arrives on stable, we can use a new `master-nightly` branch as the main development branch temporarily, then merge it in one go later.

**Comment by jamesmunns** [comment]
Hey @hannobraun, I'll try and get the examples running. Admittedly, I haven't run them before, so this will be an interesting test :)

**Comment by jamesmunns** [comment]
Hey Hanno,

After some work, I was able to get these running, both with the original master, as well as with the changes in this branch.

Unfortunately, I can only get these to work in debug builds. With release builds, I never get a ranging message (on master or with these changes). I'll open up a separate issue for that, but I have the sneaking suspicion optimization has bitten us again.

**Comment by hannobraun** [comment]
Thank you, James!

So unless I'm missing something, all that's left to do here is wait for the nrf52-hal and edition releases (and update the dependency version once they arrive).

**Comment by hannobraun** [comment]
The edition release is here and Travis is all green again. This pull request remains blocked on the `0.6.0` release of `nrf52832-hal`.

File: src/util.rs [hunk 1]
```diff
@@ -49,6 +48,10 @@ macro_rules! block_timeout {
                         unreachable!(),
                 }
 
+                // ::core::sync::atomic::compiler_fence(
+                //     ::core::sync::atomic::Ordering::AcqRel
+                // );
+
```
Comment (hannobraun) [inline src/util.rs]: What's up with the commented compiler fence? Could be removed, no?

File: src/util.rs [hunk 2]
```diff
@@ -88,6 +91,10 @@ macro_rules! repeat_timeout {
...
```
Comment (hannobraun) [inline src/util.rs]: Same

**Comment by hannobraun** [comment]
Thank you!

**Pull request merged**